app/models/miq_server/role_management.rb

```diff
@@ -153,11 +156,13 @@ def server_role_names=(roles)
         removes = ServerRole.where(:name => (current - desired))
         server_roles.delete(removes) unless removes.empty?
 
-        adds    = ServerRole.where(:name => (desired - current))
-        adds.each do |r|
-          assign_role(r)
-          deactivate_roles(r.name)
-        end unless adds.empty?
+        adds = ServerRole.where(:name => (desired - current))
+        unless adds.empty?
```

Comment:
Interesting...I'm curious why this was in the original code at all... `.each` will effectively be a no-op anyway.  Not for this PR, but this line can be removed in a follow-up

Reply:
Noted.